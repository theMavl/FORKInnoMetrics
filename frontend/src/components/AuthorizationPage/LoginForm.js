import React from 'react'
import { Link } from 'react-router-dom'
import { Field } from 'redux-form'
import Button from '../Button'
import Input from '../Input'
import { email, required } from '../../helpers/formValidators'
import styles from './style.css'
import Spinner from "../Spinner";

class LoginForm extends React.Component {
  onSubmit = () => {
    this.props.login()
  }

  render () {
    return (
      <form className={styles.form}
            onSubmit={this.props.handleSubmit(this.onSubmit)}
      >
        <h1 className={styles.msgTitle}>Login</h1>
        <Field
          name='email'
          component={Input}
          label='Email'
          placeholder='Email'
          type='email'
          validate={[required, email]}
          disabled={this.props.activeRequest}
        />
          {this.props.activeRequest && <Spinner/>}
        <Field
          name='password'
          component={Input}
          label='Password'
          placeholder='Password'
          type='password'
          validate={required}
          disabled={this.props.activeRequest}
        />
        <Button name='Login'
                type='submit'
                style={{width: '100%'}}
                disabled={this.props.submitDisabled}
        />
        <p className={styles.formError}>{this.props.formError}</p>
        <p className={styles.suggestion}>Don&#39;t have account yet?&nbsp;
          <Link to='/register'>Register</Link>
        </p>
      </form>
    )
  }
}

export default LoginForm