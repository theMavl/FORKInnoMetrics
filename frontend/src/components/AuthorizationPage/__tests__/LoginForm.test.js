import React from 'react'
import createStore from 'redux-mock-store'
import { Provider } from 'react-redux'
import { shallow } from 'enzyme'
import LoginForm from '../LoginForm'

describe('Test LoginForm component', () => {

  it('Test shallow render LoginForm component', () => {

    const props = {}
    const mockStore = createStore([])
    const store = mockStore({})

    const wrapper = shallow(
      <Provider store={store}>
        <LoginForm props={props} />
      </Provider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})