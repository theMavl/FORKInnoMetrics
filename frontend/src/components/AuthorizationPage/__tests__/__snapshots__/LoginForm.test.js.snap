// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginForm component Test shallow render LoginForm component 1`] = `
<LoginForm
  props={Object {}}
/>
`;
